### ISSUE TITLE
Cache doesn't honor "force" option during generate

### ISSUE THREAD

**Issue by CITguy** [hexojs/hexo#2700]
I have an external script to watch for changes to ANY file in `source/` (including skipped, underscore files) and force a Hexo generate when that happens. 

```javascript
hexo.call('generate', { force: true })
```

I've also got a custom renderer defined for LESS compilation.

```javascript
hexo.extend.renderer.register('less', 'css', function() { 
    /* ... */
});
```

Given the following setup:
```less
// source/styles/myStyles.less
@import '_partial';

// source/styles/_partial.less
.foo { 
  content: 'bar';
}
```
I would expect `public/styles/myStyles.css` to be regenerated with updated content any time `_partial.less` changes.

-----

When I save `_partial.less`, my script triggers a forced generation. However, the `Cache` model seems to skip `myStyles.less` because neither its contents, nor its modified time have changed.

Please update so that `Cache` will honor the `force` option passed to the [`generateConsole()`](https://github.com/hexojs/hexo/blob/5234c4a85dc6cd418e9a1c169e43de169cf98e95/lib/plugins/console/generate.js#L16) function.

Currently, the only way around this is to restart my development script so that it generates all files as if it were the first generate.

-----
https://github.com/hexojs/hexo/blob/5234c4a85dc6cd418e9a1c169e43de169cf98e95/lib/models/cache.js#L36-L41

https://github.com/hexojs/hexo/blob/5234c4a85dc6cd418e9a1c169e43de169cf98e95/lib/models/cache.js#L52-L57

**Comment by tea3** [comment]
@CITguy 
`Cache` in hexo watching content updates. Also, general usage don't include style sheets including LESS etc in the source directory.

Normally it is recommended to put the LESS file in the theme directory. In this usage, I think that it will monitor files update with hexo-render-less etc.

---

If that is inevitable, I suggest that you create a plugin to edit Cache data like [my Plugin](https://github.com/tea3/hexo-include-markdown).

- I'm monitoring the markdown change with the following code. I think this can also be applied to LESS file.

https://github.com/tea3/hexo-include-markdown/blob/master/lib/orverwriteCache.js#L52

- After finding the file's change, it will edit the cache to determine the update.

https://github.com/tea3/hexo-include-markdown/blob/master/lib/orverwriteCache.js#L161

**Comment by CITguy** [comment]
Yeah, the fun thing about my setup is that I'm using Hexo to build out documentation for a design system. So, I've got Pages documenting each component and I'm using the Posts for release notes. My script doesn't use the Hexo server, but uses Browsersync to serve up the `public/` directory. Hexo correctly generates content on deploy, but this Cache issue is frustrating while I'm trying to develop locally.
